GenericList/string_list.cpp

```diff
@@ -56,50 +104,110 @@ std::string List_String::ToString() {
 	return result;
 }
 
-
-bool List_String::Add(std::string new_val) {
-	if (capacity < count + 1) {
-		Capacity(capacity == 0 ? 1 : capacity * 2); //double array size
+std::string* List_String::Find(const std::string& val) {
+	for (int i = 0; i < count; i++) {
+		if (data[i] == val) return data+i; //TODO: research why &data[i] might get overloaded
 	}
-	data[count++] = new_val;
-	return true;
+	return nullptr;
 }
 
-bool List_String::Contains(std::string val) {
-	return IndexOf(val) != -1;
+
+const std::string* List_String::Find(const std::string& val) const {
+	for (int i = 0; i < count; i++) {
+		if (data[i] == val) return data + i; //TODO: research why &data[i] might get overloaded
+	}
+	return nullptr;
 }
 
-bool List_String::RemoveAt(int index) {
-	if (index < 0) return false;
-	if (index >= count) return false;
-	//allowed setting: index = [0, count-1], count > 0 (data is initialized)
 
-	std::memmove(data + index, data + index + 1, count - index - 1);
-	count--;
-	return true;
+void List_String::RemoveAt(int index) {
+	if (index < 0 || index >= count) throw std::out_of_range(string("Cannot remove element at out_of_range index: ") + to_string(index) + string("."));
+	
+	std::move(data + index + 1, data + count, data + index);
+	data[--count].~string();
 }
 
 
-int List_String::IndexOf(std::string val) {
-	for (int i = 0; i < count; i++) {
-		if (data[i] == val) return i;
+
+size_t List_String::Remove(const std::string& val) {
+	//Lazy, simpler way - when changing behaviour, just change it in RemoveIf:
+	/*
+	 * return RemoveIf([](const auto& e) { return e == val; } );
```

Comment:
Yea, just use this, less to change if you need to change things. also capture `val` by reference
```c++
	 return RemoveIf([&](const auto& e) { return e == val; } );
```